lib/matplotlib/cbook/__init__.py

```diff
@@ -1557,24 +1557,26 @@ def get_siblings(self, a):
 
 
 def simple_linear_interpolation(a, steps):
-    if steps == 1:
-        return a
-
-    steps = int(np.floor(steps))
-    new_length = ((len(a) - 1) * steps) + 1
-    new_shape = list(a.shape)
-    new_shape[0] = new_length
-    result = np.zeros(new_shape, a.dtype)
-
-    result[0] = a[0]
-    a0 = a[0:-1]
-    a1 = a[1:]
-    delta = ((a1 - a0) / steps)
-    for i in range(1, steps):
-        result[i::steps] = delta * i + a0
-    result[steps::steps] = a1
+    """
+    Resample an array with ``steps - 1`` points between each original point.
 
-    return result
+    Parameters
+    ----------
+    a : shape ``(N, ...)`` array-like
```

Comment:
I think this pretty much requires an array; it would fail with a list, and I think it would ignore the mask in a masked array.
Another question for my information: is this way of including shape info documented as standard, either in numpydoc or in our docs?   If not, it might be easier to read and more concise with "array, (N, ...)".  That way the kind of object comes first, followed by the additional shape info.

Reply:
A quick grep suggests that `array, shape (...)` is more common in the docs (and I agree it is more legible).
The previous implementation already required `a` to be an array so I don't think that's a problem.